frontend/rolecall/src/app/api/cast_api.service.ts

```diff
@@ -82,278 +84,318 @@ export type OneCastResponse = {
-                      return {
-                        uuid: String(rawMem.userId),
-                        position_number: rawMem.order
-                      }
-                    }));
+    const header = await this.headerUtil.generateHeader();
+    return this.http.get<AllRawCastsResponse>(
+        environment.backendURL + 'api/cast', {
+              headers: header,
+              observe: 'response',
+              withCredentials: true
+            })
+        .toPromise()
+        .catch((errorResp) => errorResp)
+        .then(
+            (resp) => this.respHandler.checkResponse<AllRawCastsResponse>(resp))
+        .then((result) => {
+          this.rawCasts = result.data;
+          const allPositions: Position[] = [];
+          Array.from(this.pieceAPI.pieces.values()).forEach(piece => {
+            allPositions.push(...piece.positions);
+          });
+          return {
+            data: {
+              casts: result.data.map(rawCast => {
+                let highestCastNumber = 0;
+                const groups: CastGroup[] = [];
+                for (const rawSubCast of rawCast.subCasts) {
+                  const foundGroup = groups.find(
+                      g => g.position_uuid === String(
+                          allPositions.find(pos =>
+                           Number(pos.uuid) === rawSubCast.positionId).uuid));
+                  if (foundGroup) {
+                    const foundGroupIndex = groups.find(
+                        g => g.position_uuid === String(allPositions.find(
+                            pos => Number(pos.uuid)
+                                   === rawSubCast.positionId).uuid)
+                             && g.group_index === rawSubCast.castNumber);
+                    if (foundGroupIndex) {
+                      foundGroupIndex.members = foundGroupIndex.members.concat(
+                          rawSubCast.members.map(rawMem => {
+                            return {
+                              uuid: String(rawMem.userId),
+                              position_number: rawMem.order
+                            };
+                          }));
+                    } else {
+                      groups.push({
+                        position_uuid: String(allPositions.find(
+                            pos => Number(pos.uuid)
+                                   === rawSubCast.positionId).uuid),
+                        group_index: rawSubCast.castNumber,
+                        members: rawSubCast.members.map(rawMem => {
+                          return {
+                            uuid: String(rawMem.userId),
+                            position_number: rawMem.order
+                          };
+                        })
+                      });
+                    }
                   } else {
                     groups.push({
-                      position_uuid: String(allPositions.find(pos => Number(pos.uuid) == rawSubCast.positionId).uuid),
```

Comment:
It seems as if the lines at 188 through 201 are the same as 174 through 187.

Reply:
Good catch, I added a TODO above to resolve this later, don't want to mess with logic in the same PR